### PR TITLE
Feat: `select ... at` for new parser & planner

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- new parser: parse `select ...  at ` 
- new planner : execute `select ... at`

## Changelog

- Improvement
- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #issue

### CODE REVIEW

File: query/src/sql/planner/binder/mod.rs [hunk 1]
```diff
@@ -18,9 +18,11 @@ pub use aggregate::AggregateInfo;
 pub use bind_context::BindContext;
 pub use bind_context::ColumnBinding;
 use common_ast::ast::Statement;
+use common_ast::ast::TimeTravelPoint;
 use common_datavalues::DataTypeImpl;
 use common_exception::ErrorCode;
 use common_exception::Result;
+use sqlparser::ast::Instant;
```
Comment (leiysky) [inline query/src/sql/planner/binder/mod.rs:25]: Maybe it's better to introduce a internal `Instant` struct.

And is it possible to use an exists struct instead?

File: query/src/sql/planner/binder/mod.rs [hunk 2]
```diff
@@ -114,10 +116,16 @@ impl<'a> Binder {
         catalog_name: &str,
         database_name: &str,
         table_name: &str,
+        travel_point: &Option<TimeTravelPoint>,
```
Comment (leiysky) [inline query/src/sql/planner/binder/mod.rs:119]:
```suggestion
        travel_point: Option<&TimeTravelPoint>,
```

File: common/ast/src/parser/query.rs [hunk 3]
```diff
@@ -153,11 +153,21 @@ pub fn aliased_table(i: Input) -> IResult<TableReference> {
...
+                travel_point,
             }
         },
     )(i)
 }
 
+pub fn travel_point(i: Input) -> IResult<TimeTravelPoint> {
+    map(
+        rule! {
+            AT ~  "(" ~ SNAPSHOT ~ "=>" ~ #literal_string ~ ")"
```
Comment (andylokandy) [inline common/ast/src/parser/query.rs:165]:
```suggestion
            AT ~ "(" ~ SNAPSHOT ~ "=>" ~ #literal_string ~ ")"
```

**Review by andylokandy — APPROVED** [review]

File: query/src/sql/planner/binder/mod.rs (hunk 1 above)
Comment (dantengsky) [reply query/src/sql/planner/binder/mod.rs:25]: `sqlparser::ast::Instant` is an new struct that we introduced to `sqlparser-rs`

https://github.com/datafuse-extras/sqlparser-rs/blob/7d74a3c5fc78f8d0c0f7e4e7e14d7058f94d8841/src/ast/query.rs#L321-L325

Shall we introduce another intermediate data structure that  `TimeTravelPoint` and `sqlparser::ast::Instant` can both convert into?

File: query/src/sql/planner/binder/mod.rs (hunk 2 above)
Comment (dantengsky) [reply query/src/sql/planner/binder/mod.rs:119]: oops, there are compilation errors after this suggestion is applied. so I revert it.

travel point is captured here:
https://github.com/datafuselabs/databend/blob/e2a4d789b12805d5d50eda755828ca2c872c2251/common/ast/src/parser/query.rs#L139
Which is of type `Option<TimeTravelPoint>`.

did I missed something here?  

Or  `Option<&T>` style is recommended in new parser/planner?

File: query/src/sql/planner/binder/mod.rs (hunk 1 above)
Comment (leiysky) [reply query/src/sql/planner/binder/mod.rs:25]: It's good to have one. We will deprecate the sqlparser crate eventually.

Comment (dantengsky) [reply query/src/sql/planner/binder/mod.rs:25]: OK

File: query/src/sql/planner/binder/mod.rs (hunk 2 above)
Comment (leiysky) [reply query/src/sql/planner/binder/mod.rs:119]: It's recommended, so you don't need to write a `&None`.

File: query/src/sql/planner/binder/mod.rs (hunk 1 above)
Comment (dantengsky) [reply query/src/sql/planner/binder/mod.rs]: pls have look at commit `Introduce NavigationPoint` [#5546572a36e36e03473f945fc31b555374c957dc](https://github.com/datafuselabs/databend/pull/5628/commits/5546572a36e36e03473f945fc31b555374c957dc)

**Review by leiysky — APPROVED** [review]

**Pull request merged**